_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+```
+
+In K8s Lens, this looks like this:
+
+{% img blog/kubernetes-jhipster-aws/lens-status-green.png alt:"Lens status green" width:"1000" %}{: .center-image }
+
+
+Another useful command is `describe` (I won’t replicate the output here). This is great for more detailed information for debugging.
+
+```
+kubectl describe pods -n demo
+```
+
+You can also tail logs of a specific pod using the following command.
+
+```
+kubectl logs <pod-name> --tail=-1 -n demo
+```
+
+To access the Eureka registry service, you need to forward the port.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Open a browser and navigate to http://localhost:8761
+
+You want to see a lot of green.
+
+{% img blog/kubernetes-jhipster-aws/registry-green.png alt:"Registry status OK" width:"1000" %}{: .center-image }
+
+If everything is good, `control-c` to stop forwarding the registry. Forward the gateway and open it.
+
+```bash
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Open http://localhost:8080
+
+You'll see the front page of the demo application.
+
+{% img blog/kubernetes-jhipster-aws/jhipster-demo-front.png alt:"JHipster front page" width:"1000" %}{: .center-image }
+
+Go to **Account** and **Sign in**.
+
+If you are already logged into Okta, you won't have to log in again. Otherwise you'll be redirected to the Okta SSO sign-in page.
+
+{% img blog/kubernetes-jhipster-aws/okta-sign-in.png alt:"Okta sign-in page" width:"400" %}{: .center-image }
+
+Once you're authenticated, you'll be directed to the authenticated front page.
+
+{% img blog/kubernetes-jhipster-aws/jhipster-authenticated.png alt:"Authenticated" width:"1000" %}{: .center-image }
+
+To verify that the application works, try and create a product and a blog. Go to the **Entities** menu and click on **Product**. Click **Create a new Product** and fill in the form with dummy values. You should be able to successfully create a new product. Do the same thing with **Entities** and **Blog**. You should be able to create a blog and a blog post.
+
+Remember that the blog application uses a Ne04j database backend and the store uses a Mongo DB backend. You're accessing these services through the public gateway application, but this is forwarding the requests to the private services and their associated databases.
+
+## Protect sensitive configuration values using `kubeseal`
+
+The microservice is working. The only thing left to do is to protect any sensitive configuration parameters that you don't want to check into a repository. In this case, the main value that needs to be protected is the Okta OIDC client secret (and the `ENCRYPT_KEY`, which, as you'll see in a moment, is used to encrypt the client secret but which itself will need to be encrypted using `kubeseal`).
+
+ In `kubernetes/registry-k8s/jhipster-registry.yml`, add an environment key-value pair (you can add it anywhere under the same `env` list that you edited earlier). Make up an encryption key. The longer the better.
```

Comment:
you can also create an uuid using `uuid` command if available on the OS, Linux has it not sure about mac/windows